### PR TITLE
search frontend: infer correct patterntype for eslint

### PR DESCRIPTION
**Pull request by rvantonder**
Go away yellow squiggles.

### CODE REVIEW

File: client/web/src/search/input/toggles/Toggles.tsx [hunk 1]
```diff
@@ -98,9 +98,9 @@ export const Toggles: React.FunctionComponent<TogglesProps> = (props: TogglesPro
         const cascadePatternTypeValue =
             settingsCascade.final &&
             !isErrorLike(settingsCascade.final) &&
-            settingsCascade.final['search.defaultPatternType']
+            (settingsCascade.final['search.defaultPatternType'] as SearchPatternType)
```
Comment (limitedmage) [inline client/web/src/search/input/toggles/Toggles.tsx:101]: You could instead use [`defaultPatternTypeFromSettings`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/util/settings.ts)

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/input/toggles/Toggles.tsx (hunk 1 above)
Comment (rvantonder) [reply client/web/src/search/input/toggles/Toggles.tsx:101]: Oh good call. Why does that function return `SearchPatternType or undefined`? Weird. Should just return a default, no?

Comment (rvantonder) [reply client/web/src/search/input/toggles/Toggles.tsx]: On second thought, I don't think this works. In this part of the code we want to unconditionally check settings, while it is conditional in that function: `if (!parseSearchURLPatternType(window.location.search)) ...`. I will try refactor for it.

Comment (rvantonder) [reply client/web/src/search/input/toggles/Toggles.tsx:101]: Nope, doesn't work to retry using the function in that file. When the helper is accessed in this toggle code, importing `settings.ts`, it tries to evaluate this toplevel function:

```
export const SITE_SUBJECT_NO_ADMIN: Pick<GQL.ISettingsSubject, 'id' | 'viewerCanAdminister'> = {
    id: window.context.siteGQLID,
    viewerCanAdminister: false,
}
```

and then tests barf with

```
 FAIL  client/web/src/nav/GlobalNavbar.test.tsx
  ● Test suite failed to run

    TypeError: Cannot read property 'siteGQLID' of undefined

      10 | /** A fallback settings subject that can be constructed synchronously at initialization time. */
      11 | export const SITE_SUBJECT_NO_ADMIN: Pick<GQL.ISettingsSubject, 'id' | 'viewerCanAdminister'> = {
    > 12 |     id: window.context.siteGQLID,
         |                        ^
      13 |     viewerCanAdminister: false,
      14 | }
      15 | 

      at Object.<anonymous> (src/util/settings.ts:12:24)
      at Object.<anonymous> (src/search/input/toggles/Toggles.tsx:17:1)
      at Object.<anonymous> (src/search/input/QueryInput.tsx:38:1)
      at Object.<anonymous> (src/search/input/interactive/FilterInput.tsx:20:1)
      at Object.<anonymous> (src/search/input/interactive/SelectedFiltersRow.tsx:2:1)
      at Object.<anonymous> (src/search/input/interactive/InteractiveModeInput.tsx:6:1)
      at Object.<anonymous> (src/nav/GlobalNavbar.tsx:23:1)
      at Object.<anonymous> (src/nav/GlobalNavbar.test.tsx:5:1)

 FAIL  client/web/src/search/input/LazyMonacoQueryInput.test.tsx
  ● Test suite failed to run

    TypeError: Cannot read property 'siteGQLID' of undefined

      10 | /** A fallback settings subject that can be constructed synchronously at initialization time. */
      11 | export const SITE_SUBJECT_NO_ADMIN: Pick<GQL.ISettingsSubject, 'id' | 'viewerCanAdminister'> = {
    > 12 |     id: window.context.siteGQLID,
         |                        ^
      13 |     viewerCanAdminister: false,
      14 | }
      15 | 

      at Object.<anonymous> (src/util/settings.ts:12:24)
      at Object.<anonymous> (src/search/input/toggles/Toggles.tsx:17:1)
      at Object.<anonymous> (src/search/input/LazyMonacoQueryInput.tsx:4:1)
      at Object.<anonymous> (src/search/input/LazyMonacoQueryInput.test.tsx:4:1)
```

Don't know how to fix this and seems like not a good use of time. I did update that function in `settings.ts` to get rid of yellow squiggles to though.

**Pull request merged**